src/components/HOC/withApi.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+const withApi = (WrappedComponent, charactersQuantity) => {
+    class WithApi extends React.Component {
+        constructor(props) {
+            super(props)
+            this.state = {
+                charactersCount: 0,
+                charactersArr: [],
+                randomIds: [],
+                test: "test works"
+            }
+        }
+        
+        getCharactersArr = (callback) =>{
+            this.getCharactersCount(() => {
+                this.getRandomIds(charactersQuantity, () => {
+                    this.getCharacters(callback);
+                })
+            }) 
+        }
+
+        getCharactersCount = (callback) => {
+            fetch(
+                `https://rickandmortyapi.com/api/character`
+            )
+                .then(res => res.json())
+                .then(res => {
+                    this.setState({
+                        charactersCount: res.info.count
+                    }, callback);
+                })
+                .catch(error => console.error('error:', error));
+        }
```

Comment:
Semicolons missing - another good reason to have eslint to check it for you :)

Reply:
linting solved all of this problems :)